pkgs/by-name/lu/luau-lsp/package.nix

```diff
@@ -0,0 +1,50 @@
+{ fetchFromGitHub
+, stdenv
+, lib
+, cmake
+, gcc9
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "luau-lsp";
+  version = "1.28.1";
+
+  src = fetchFromGitHub {
+    owner = "JohnnyMorganz";
+    repo = "luau-lsp";
+    rev = finalAttrs.version;
+    hash = "sha256-5+m5tLX0DRHrV6yI/SXc39orpAu1DlBmpNQ/rt7LUjQ=";
+    fetchSubmodules = true;
+  };
+
+  # Build fails if this is false
+  dontUseCmakeConfigure = true;
+
+  nativeBuildInputs = [ cmake ]
+    ++ lib.optional stdenv.isLinux gcc9;
+
+  buildPhase = ''
+    runHook preBuild
+    mkdir build && cd build
+    cmake .. -DCMAKE_BUILD_TYPE=Release
+    cmake --build . --target Luau.LanguageServer.CLI --config Release
+    runHook postBuild
```

Comment:
See above thread, too. We should avoid as much as possible duplicating the cmake hook